logger/logger.go

```diff
@@ -13,11 +13,19 @@ type causer interface {
 
 var GlobalFields = logrus.Fields{}
 
-type Logger func(Valuer, error) *logrus.Entry
+type Logger func(Valuer, ...error) *logrus.Entry
 
 func New(name string) Logger {
-	return func(ctx Valuer, err error) *logrus.Entry {
-		return ContextLog(ctx, err, nil).WithField("component", name)
+	return func(ctx Valuer, err ...error) *logrus.Entry {
+		switch len(err) {
+		case 0:
+			return ContextLog(ctx, nil, nil).WithField("component", name)
+		case 1:
+			return ContextLog(ctx, err[0], nil).WithField("component", name)
+		default:
+			ContextLog(ctx, nil, nil).WithField("component", name).Errorf("%d errors provided, expected 0 or 1", len(err))
+			return ContextLog(ctx, nil, nil).WithField("component", name)
```

Comment:
```suggestion
			return ContextLog(ctx, err[0], nil).WithField("component", name)
```
We know there’s at least 1, so might as well log it

Reply:
I don't think it will happen often, I just hate a signature that lies, and it's pretty easy to avoid here.  I'd rather see keeping the `log(ctx, nil)` form, or true variadicity. (Though really I wish go supported optional arguments properly)

Reply:
I think the current PR is fine and we already have that pattern for tags in the metrics pkg. 

That being said, if multiples become `error`, `error2`, `error3`, I’m fine with that, as long as the first one is still `error` (not `error1`) so we can splunk a reliable name

Reply:
> we already have that pattern for tags in the metrics pkg

I know, and at some point I'd like to get rid of it.

Reply:
Hmm, the problem about passing all errors down to `ContextLog` is that the signature would need to change. Currently:

```
ContextLog(ctx Valuer, err error, entry *logrus.Entry) *logrus.Entry
```

But needs to change to one of:

```
ContextLog(ctx Valuer, err []error, entry *logrus.Entry) *logrus.Entry
ContextLog(ctx Valuer, entry *logrus.Entry, err ...error) *logrus.Entry
```

(since variadic arguments need to be last).

I went with `ContextLog(ctx Valuer, err []error, entry *logrus.Entry) *logrus.Entry`, which I think is nicest.

@lavoiesl @lugray please have another look.